account_financial_report/wizard/general_ledger_wizard.py

```diff
@@ -180,28 +183,24 @@ def _check_company_id_date_range_id(self):
                     _('The Company in the General Ledger Report Wizard and in '
                       'Date Range must be the same.'))
 
-    @api.onchange('receivable_accounts_only', 'payable_accounts_only')
-    def onchange_type_accounts_only(self):
-        """Handle receivable/payable accounts only change."""
-        if self.receivable_accounts_only or self.payable_accounts_only:
-            domain = [('company_id', '=', self.company_id.id)]
-            if self.receivable_accounts_only and self.payable_accounts_only:
-                domain += [('internal_type', 'in', ('receivable', 'payable'))]
-            elif self.receivable_accounts_only:
-                domain += [('internal_type', '=', 'receivable')]
-            elif self.payable_accounts_only:
-                domain += [('internal_type', '=', 'payable')]
-            self.account_ids = self.env['account.account'].search(domain)
+    @api.onchange('account_type_ids')
+    def _onchange_account_type_ids(self):
+        if self.account_type_ids:
+            self.account_ids = self.env['account.account'].search([
+                ('user_type_id', 'in', self.account_type_ids.ids)])
         else:
             self.account_ids = None
 
     @api.onchange('partner_ids')
     def onchange_partner_ids(self):
         """Handle partners change."""
         if self.partner_ids:
-            self.receivable_accounts_only = self.payable_accounts_only = True
+            self.account_type_ids = self.env['account.account.type'].search([
+                ('type', 'in', ['receivable', 'payable'])])
```

Comment:
the same remark about the company

Reply:
Account Types have no company. In this case it's not necessary